trail/upload/process_uploads/header_standardizer.py

```diff
@@ -0,0 +1,329 @@
+        cornerRa = cornerSkyCoord.ra.to(u.deg)
+        cornerDec = cornerSkyCoord.dec.to(u.deg)
+
+        unitSphereCenter = np.array([
+            np.cos(centerDec) * np.cos(centerRa),
+            np.cos(centerDec) * np.sin(centerRa),
+            np.sin(centerDec)
+        ])
+
+        unitSphereCorner = np.array([
+            np.cos(cornerDec) * np.cos(cornerRa),
+            np.cos(cornerDec) * np.sin(cornerRa),
+            np.sin(cornerDec)
+        ])
+
+        unitRadius = np.linalg.norm(unitSphereCenter-unitSphereCorner)
+        standardizedWcs["wcs_radius"] = unitRadius
+
+        standardizedWcs["wcs_center_x"] = unitSphereCenter[0]
+        standardizedWcs["wcs_center_y"] = unitSphereCenter[1]
+        standardizedWcs["wcs_center_z"] = unitSphereCenter[2]
+
+        standardizedWcs["wcs_corner_x"] = unitSphereCorner[0]
+        standardizedWcs["wcs_corner_y"] = unitSphereCorner[1]
+        standardizedWcs["wcs_corner_z"] = unitSphereCorner[2]
+
+        return standardizedWcs
+
+    # wow, do not flip these two decorators around...
+    @classmethod
+    @abstractmethod
+    def canStandardize(self, header, **kwargs):
+        """Returns `True` when the standardizer knows how to handle given
+        upload.
+
+        Parameters
+        ----------
+        header : `object`
+             The header, Astropy HDU and its derivatives.
+        \**kwargs : `dict`
+            Additional keyword arguments
+
+        Keyword arguments
+        -----------------
+        filename : `str`
+            Name of the file from which the HDU was read from, sometimes can encode
+            additional metadata.
+
+        Returns
+        -------
+        canProcess : `bool`
+            `True` when the processor knows how to handle uploaded file and
+            `False` otherwise
+        """
+        raise NotImplemented()
+
+    @classmethod
+    def getStandardizer(cls, header, **kwargs):
+        """Get the standardizer class that can handle given header.
+
+        Parameters
+        ----------
+        header : `object`
+             The header, Astropy HDU and its derivatives.
+        \**kwargs : `dict`
+            Additional keyword arguments
+
+        Keyword arguments
+        -----------------
+        filename : `str`
+            Name of the file from which the HDU was read from, sometimes can encode
+            additional metadata.
+
+        Returns
+        -------
+        standardizerCls : `cls`
+            Standardizer class that can process the given upload.`
+
+        Raises
+        ------
+        ValueError
+            None of the registered processors can process the  upload.
+        """
+        standardizers = []
+        for standardizer in cls.standardizers.values():
+            if standardizer.canStandardize(header):
+                standardizers.append(standardizer)
+        standardizers.sort(key=lambda standardizer: standardizer.priority, reverse=True)
```

Comment:
Any way to rewrite this without a lambda? I just despise them is all.

Reply:
Sure, it would look something like:

```
# ... for loop ...
def sort_by_priority(standardizer):
    return standardizer.priority
standardizers.sort(key=sort_by_priority, reverse=True)
```

if you're fine with that...

Reply:
I can know that's what it's doing if I stare at it for 10 min. If it doesn't sacrifice speed, I prefer the clarity - thanks 😁